tests/general/CMakeLists.txt

```diff
@@ -348,20 +348,20 @@ else ()
 endif ()
 
 #===== pio_iosystems_test2 =====
-# add_executable (pio_iosystem_tests2 EXCLUDE_FROM_ALL
-#     pio_iosystem_tests2.F90
-#     ${CMAKE_CURRENT_SOURCE_DIR}/util/pio_tutil.F90)
-# target_link_libraries (pio_iosystem_tests2 piof)
-# add_dependencies (tests pio_iosystem_tests2)
+add_executable (pio_iosystem_tests2 EXCLUDE_FROM_ALL
+    pio_iosystem_tests2.F90
+    ${CMAKE_CURRENT_SOURCE_DIR}/util/pio_tutil.F90)
+target_link_libraries (pio_iosystem_tests2 piof)
+add_dependencies (tests pio_iosystem_tests2)
 
-# if (PIO_USE_MPISERIAL)
-#     add_test(NAME pio_iosystem_tests2
-#         COMMAND pio_iosystem_tests2)
-#     set_tests_properties(pio_iosystem_tests2
-#         PROPERTIES TIMEOUT ${DEFAULT_TEST_TIMEOUT})
-# else ()
-#     add_mpi_test(pio_iosystem_tests2
-#         EXECUTABLE ${CMAKE_CURRENT_BINARY_DIR}/pio_iosystem_tests2
-#         NUMPROCS 4
-#         TIMEOUT ${DEFAULT_TEST_TIMEOUT})
-# endif ()
+if (PIO_USE_MPISERIAL)
+    add_test(NAME pio_iosystem_tests2
+        COMMAND pio_iosystem_tests2)
+    set_tests_properties(pio_iosystem_tests2
+        PROPERTIES TIMEOUT ${DEFAULT_TEST_TIMEOUT})
+else ()
+    # add_mpi_test(pio_iosystem_tests2
```

Comment:
Howdy @jayeshkrishna!

CI means that you can't leave broken tests turned on in the build, or else all CI builds show as broken.

The point of CI is that it is automated, not that I go look at what failed every time and say "that's OK." The idea is that I can tell at a glance if code I've just checked in caused any tests to fail.

My CI page builds the library in various configurations (ex with and without async, with and without logging, etc.), and so there are a number of builds that show up as broken when there is a failing test.

The way to track known bugs is in the github issues list, not by leaving in failing tests. 

There is a good description of the git workflow here: https://www.atlassian.com/git/tutorials/comparing-workflows/centralized-workflow (we are using what is called "Feature Branch Workflow")

From this document: 

> The core idea behind the Feature Branch Workflow is that all feature development should take place in a dedicated branch instead of the master branch. This encapsulation makes it easy for multiple developers to work on a particular feature without disturbing the main codebase. It also means the master branch will never contain broken code, which is a huge advantage for continuous integration environments.

We have to share the master branch, so we need to leave it in a working state at all times.

I have commented out this test on master. When you take your next branch, simply uncomment the test on your branch, and leave it commented out on master, so we can continue to use CI. When your merge your branch with the fix, you will also merge the uncommenting of the test, and so all will be right with the world.

Your fixes, and your working test, will be merged into master at the same time, so that master will always yield a stable build that passes all tests. 

Thanks!